### PR TITLE
[EOC-402] Refactor events emiting from actions to controllers for list

### PR DESCRIPTION
**Pull request by adamklepacz**
There is some commented code on this branch, that disabled sockets logic for 
marking item as done /unhandled('updateItem' on backend, toggleItem action on front-end).
This action and controller needs to be refactored from tooglleItem to 
`markAsDone(), markAsUnhandled(),` because toggling causes problems right now, and socket functionality doesn't work properly. 

This will be fixed and refactored on **EOC-404 [IMPROVEMENT] Refactor events emitting from actions to controllers for ITEMS**

### CODE REVIEW

File: server/sockets/list.js [hunk 1]
```diff
@@ -782,113 +776,93 @@ const archiveList = (socket, dashboardClients, cohortClients, listClients) =>
...
-      });
-  });
+  io.sockets
+    .to(`sack-${listId}`)
+    .emit(ListActionTypes.DELETE_AND_REDIRECT, data);
 
-const deleteList = (socket, dashboardClients, cohortClients) =>
-  socket.on(ListActionTypes.DELETE_SUCCESS, data => {
-    const { listId, cohortId } = data;
+  dashboardClients.forEach(client => {
```
Comment (marekrozmus) [inline server/sockets/list.js]: could be one line:
`...forEach(({ socketId }) => ...`

File: server/sockets/index.js [hunk 2]
```diff
@@ -70,16 +71,18 @@ const socketListenTo = server => {
...
+  }
 
-  const cohortViewClients = new Map();
-  const allCohortsViewClients = new Map();
-  const dashboardViewClients = new Map();
-  const listViewClients = new Map();
-  const cohortClientLocks = new Map();
-  const itemClientLocks = new Map();
-  const listClientLocks = new Map();
+  return socketInstance;
```
Comment (alemikolo) [inline server/sockets/index.js]: could we return socket instead of socketInstance to not use io.socket everywhere but only socket?

Comment (adamklepacz) [reply server/sockets/index.js]: I've tried to export just `return socketInstance.sockets ` but it doesn't work.

File: server/controllers/list.js [hunk 3]
```diff
@@ -449,21 +477,47 @@ const updateListById = (req, resp) => {
...
+          archiveList(
+            socketInstance,
+            dashboardClients,
+            cohortClients,
+            listClients
+          )(data);
+        }
+
+        listActivity = ActivityType.LIST_RESTORE;
+        restoreList(
```
Comment (Ciunkos) [inline server/controllers/list.js]: Please `return` from the actions so no next actions are fired after them.

File: server/controllers/list.js [hunk 4]
```diff
@@ -719,9 +784,10 @@ const removeOwnerRole = (req, resp) => {
...
 
-      resp.send({
-        message: 'User has no owner role.'
-      });
+      const data = { listId, userId };
+
+      removeOwnerRoleInList(socketInstance, listClients)(data);
+      resp.send();
 
       saveActivity(
```
Comment (Ciunkos) [inline server/controllers/list.js]: Please move it to the `fireAndForget` handler.

File: server/controllers/list.js [hunk 5]
```diff
@@ -991,6 +1072,13 @@ const updateListItem = (req, res) => {
         editedItemActivity = isOrdered
           ? ActivityType.ITEM_DONE
           : ActivityType.ITEM_UNHANDLED;
+
+        // THIS METHOD NEED TO RECOGNIZE CLIENT SOMEHOW,
```
Comment (Ciunkos) [inline server/controllers/list.js]: Please add `// TODO:` to make searching for them easier.

File: server/controllers/list.js (hunk 3 above)
Comment (Ciunkos) [reply server/controllers/list.js]: Please `then` all actions and return responses.

File: server/controllers/list.js [hunk 6]
```diff
@@ -1157,7 +1245,15 @@ const changeType = (req, resp) => {
...
+
+      changeListType(
+        socketInstance,
+        dashboardClients,
+        cohortClients,
+        listClients
+      )(data);
+      resp.send({ members, type });
 
       saveActivity(
```
Comment (Ciunkos) [inline server/controllers/list.js]: `const fireAndForget = promise => promise.catch((error) => console.error(error));` `fireAndForget( saveActivity(...))`

File: server/controllers/list.js [hunk 7]
```diff
@@ -1157,7 +1245,15 @@ const changeType = (req, resp) => {
...
       );
 
-      resp.send({ members, type, removedViewers });
+      const data = { listId, type, removedViewers };
+
+      changeListType(
+        socketInstance,
+        dashboardClients,
+        cohortClients,
+        listClients
```
Comment (Ciunkos) [inline server/controllers/list.js]: `return changeListType(...).then(() => ({ members, type })`. `.then(payload => resp.send(payload))`

File: server/controllers/list.js [hunk 8]
```diff
@@ -719,18 +824,22 @@ const removeOwnerRole = (req, resp) => {
...
+        saveActivity(
+          ActivityType.LIST_SET_AS_MEMBER,
+          ownerId,
+          null,
+          sanitizedListId,
+          list.cohortId,
+          userId
+        )
       );
+      resp.send();
```
Comment (Ciunkos) [inline server/controllers/list.js]: Please add a new line for consistency.

File: server/index.js [hunk 9]
```diff
@@ -7,8 +7,18 @@ const passport = require('passport');
 const cookieParser = require('cookie-parser');
 const session = require('express-session');
 const path = require('path');
+/* eslint-disable import/order */
```
Comment (Ciunkos) [inline server/index.js]: Why this rule is disabled? Wouldn't it be better to just sort the imports?

Comment (adamklepacz) [reply server/index.js]: Added as EOC-433 task in JIRA

**Pull request merged**